databind/src/main/java/tech/ydb/yoj/databind/schema/reflect/StdReflector.java

```diff
@@ -27,7 +27,7 @@ public final class StdReflector implements Reflector {
     public static final Reflector instance = new StdReflector(List.of(
             RecordType.FACTORY,
             KotlinDataClassTypeFactory.INSTANCE,
-            SelectorPojoFactory.INSTANCE,
+            StdReflector.strictMode ? StrictPojoFactory.INSTANCE : LegacyPojoFactory.INSTANCE,
```

Comment:
this check is made TOO EARLY, `StdReflector` class first gets initialized AND ONLY THEN you are able to call `StdReflector.enableStrictMode()`... but it's TOO LATE ALREADY for the correct PojoFactory to be picked